presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -307,6 +310,11 @@ public QueryInfo createQuery(SessionSupplier sessionSupplier, String query)
         Statement statement;
         try {
             session = sessionSupplier.createSession(queryId, transactionManager, accessControl, sessionPropertyManager);
+            if (query.length() > maxQueryLength) {
+                int queryLength = query.length();
```

Comment:
@electrum  When throwing exception in line 316,  we will report the length of the original query (instead of truncated one, which will always be ` maxQueryLength`).